### PR TITLE
Release/2.3.0

### PR DESCRIPTION
**Pull request by mbercx**

### CODE REVIEW

File: .github/workflows/update_changelog.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        # Add the commit hash (short) to link to the changelog
+        commit = commit.strip("'")
+        hash_short, hash_long, message = commit.split('|', maxsplit=2)
+        message += f' [[{hash_short}](https://github.com/aiidateam/qe-tools/commit/{hash_long})]'
+        changelog_message += f'\n* {message}'
+
+    with Path('CHANGELOG.md').open('w', encoding='utf8') as handle:
+        handle.write(changelog_message + '\n\n' + current_changelog)
+
```
Comment (t-reents) [inline .github/workflows/update_changelog.py:86]:
```suggestion
    changelog_header = '# Change log\n\n'
    with Path('CHANGELOG.md').open('w', encoding='utf8') as handle:
        handle.write(changelog_header + current_changelog.replace(changelog_header, changelog_message + '\n\n'))
```

File: .github/workflows/update_changelog.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    commits = subprocess.run(['git', 'log', "--pretty=format:'%h|%H|%s'", f'{latest_tag}..origin/main'],
+                             capture_output=True,
+                             check=True,
+                             encoding='utf-8').stdout
+
+    pr_pattern = re.compile(r'\(\S(?P<pr_number>\d+)\)')
+
+    changelog_message = f'## v{__version__}\n' + DEFAULT_CHANGELOG_SECTIONS
+
+    for commit in commits.splitlines():
```
Comment (t-reents) [inline .github/workflows/update_changelog.py:69]:
```suggestion
    release_date = date.today()
    changelog_message = f'## `v{__version__}` - {release_date}\n' + DEFAULT_CHANGELOG_SECTIONS
```

**Review by t-reents — APPROVED** [review]
I know that this is a kind of standard function (also used in AiiDA-QE) that requires some manual adjustments, anyway. However, I realized that the title `Changelog` is simply overwritten and that the date is missing next to the new version. Therefore, I added some small suggestions to also respect these aspects in the generated template.

**Comment by mbercx** [comment]
Thanks @t-reents! I simply removed the title in https://github.com/aiidateam/qe-tools/commit/11f7a9227ec5d4873375147a43fb107ad095a5e2, that simplifies the code and I don't see a reason to keep it. ^^

Going to merge once the tests pass. 🚀

**Pull request merged**